### PR TITLE
Remove ./bin/trial

### PR DESCRIPTION
**Pull request by rodrigc**
Needed by https://github.com/twisted/twisted/pull/504

### CODE REVIEW

File: services/buildbot/master/twisted_factories.py [hunk 1]
```diff
@@ -523,7 +523,7 @@ class TwistedVirtualenvReactorsBuildFactory(TwistedBaseFactory):
     treeStableTimer = 5*60
 
     def __init__(self, source, RemovePYCs=RemovePYCs, python="python",
-                 trial="./bin/trial", virtualenv_module="virtualenv",
+                 trial="-m twisted.trial", virtualenv_module="virtualenv",
```
Comment (adiroiban) [inline services/buildbot/master/twisted_factories.py]: in this context, do we still need the trial argument at all?

I am assuming that all tests will now run with `-m twisted.trial`

**Comment by adiroiban** [comment]
I think that we need this, but instead of focusing on removing trial, we should focus on having all builders using tox to get a similar run as on travis/appeveyor. 

In this way we will also get rid of trial.

We can start with removing a few builders like the one targeted in this PR.

With this change we should also cleanup the code which is no longer used ... ex remove TwistedVirtualenvReactorsBuildFactory

File: services/buildbot/master/twisted_factories.py (hunk 1 above)
Comment (rodrigc) [inline services/buildbot/master/twisted_factories.py]: Maybe not, but I don't have access to the builders, so I am only guessing
at how the buildbot logic works by looking at it.

**Comment by rodrigc** [comment]
trial needs to go ASAP, because people still try to check out the code and use it in the tree.
Because of new dependencies on constantly and moving of the source code to src,
it no longer works the way it use to in the tree, leading to confusion.

I wanted to remove trial before, but was opposed during code review, due to the code in the buildbots which use it.

**Comment by adiroiban** [comment]
we can run buildmaster/buildslave in a local vagrant VM for testing/development ... and this is how all buildbot changes should be done before deployment 

here is some documentation https://github.com/twisted-infra/braid#vagrant

all buildbot configuration is public and is defined here https://github.com/twisted-infra/braid/blob/master/services/buildbot/master/

**Comment by rodrigc** [comment]
It's not clear to me how to use this vagrant setup to test buildbot.  I tried following the steps
and was prompted for a passphrase when running commands like **fab config.vagrant buildbot.restart**.  Also, the vagrant instructions do not make it obvious how to test changes to Windows buildbot configurations.

**Comment by adiroiban** [comment]
testing windows slaves is not yet supported

if if asked your for a password, then the vagrant vm was not set up ok... can you do vagrant ssh to the machine?

there are a couple of tickets targeting the removal of non-tox builders:

* #212 - nomodules run
* #214 - windows bdist builder

but I think that first we should have https://twistedmatrix.com/trac/ticket/8544 this implemented and merged

**Comment by adiroiban** [comment]
so on https://twistedmatrix.com/trac/ticket/8544 due to https://github.com/pypa/virtualenv/issues/929 I have a bad time implementing tox/venv on Windows

and this branch is doing the same... trying to run tox/venv on Windows to get rid of bin/trial

we might want to change strategy and remove the windows builds from buildbot and only have them on appveyor ... but appveyor is windows2012r2 ... and we are no generating the release wheel on win7 (win2008) ... so not sure if this will have an impact 

I will do one more push for the venv --clear workaround on windows

**Comment by glyph** [comment]
Is this still relevant?

**Comment by rodrigc** [comment]
It looks like all the builds have been converted to use Tox in master.cfg.
The code which references ./bin/trial can probably go away because those steps are not used.

**Pull request closed without merging**